src/utils/SupersetUtils.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+export const buildSupersetUrlParams = (reportConfig, context, contextType = 'scenario') => {
+  const { currentScenarioData, visibleScenarios } = context;
+  if (!reportConfig?.dynamicFilters) return {};
+
+  const preselectFilters = {};
+
+  for (const filter of reportConfig.dynamicFilters) {
+    const { chartId, column, values } = filter;
+    let resolvedValues;
+
+    if (contextType === 'scenario') {
+      if (values === 'csmSimulationRun') resolvedValues = [currentScenarioData?.lastRun?.csmSimulationRun];
+      else if (values === 'lastRunId') resolvedValues = [currentScenarioData?.lastRunId];
+    } else if (contextType === 'dashboards') {
+      if (values === 'visibleScenariosCsmSimulationRunsIds')
+        resolvedValues = visibleScenarios?.map((s) => s.lastRun?.csmSimulationRun);
+      else if (values === 'visibleScenariosSimulationRunsIds') resolvedValues = visibleScenarios?.map((s) => s.runId);
+    }
```

Comment:
I'm not sure we need the scenario vs. dashboard context distinction, because all dynamic filters can be used in both Scenario and Dashboard views